diff --git a/libraries/mi/src/mi_uncompress.c b/libraries/mi/src/mi_uncompress.c
index 167e12b..23593ac 100644
--- a/libraries/mi/src/mi_uncompress.c
+++ b/libraries/mi/src/mi_uncompress.c
@@ -57,20 +57,9 @@ asm void MI_UnpackBits (register const void * srcp, register void * destp, regis
 
 asm void MI_UncompressLZ8 (register const void * srcp, register void * destp)
 {
-    stmfd sp !, {r4 - r7, lr}
+    stmfd sp !, {r4 - r6, lr}
     ldr r5, [r0], #4
     mov r2, r5, lsr #8
-    mov r7, #0
-    tst r5, #0x0F
-#ifdef SP1P3_BUG_FOR_CONDITIONAL_ASM_INSTRUCTIONS
-    bne @movne1
-    b @movne2
-@movne1:
-    mov r7, #1
-@movne2:
-#else
-    movne r7, #1
-#endif
 @21:
     cmp r2, #0
     ble @26
@@ -88,44 +77,10 @@ asm void MI_UncompressLZ8 (register const void * srcp, register void * destp)
     b  @25
 @23:
     ldrb r5, [r0, #0]
-    cmp r7, #0
-#ifdef SP1P3_BUG_FOR_CONDITIONAL_ASM_INSTRUCTIONS
-    beq @moveq3
-    b @moveq4
-@moveq3:
     mov r6, #3
-@moveq4:
-#else
-    moveq r6, #3
-#endif
-    beq @23_2
-    tst r5, #0xE0
-#ifdef SP1P3_BUG_FOR_CONDITIONAL_ASM_INSTRUCTIONS
-    bne @movne5
-    b @movne6
-@movne5:
-    mov r6, #1
-@movne6:
-#else
-    movne r6, #1
-#endif
-    bne @23_2
-    add r0, r0, #1
-    and r6, r5, #0xf
-    mov r6, r6, lsl #4
-    tst r5, #0x10
-    beq @23_1
-    mov r6, r6, lsl #8
-    ldrb r5, [r0], #1
-    add r6, r6, r5, lsl #4
-    add r6, r6, #0x100
-@23_1:
-    add r6, r6, #0x11
-    ldrb r5, [r0, #0]
-@23_2:
     add r3, r6, r5, asr #4
-    add r0, r0, #1
-    and r5, r5, #0xf
+    ldrb r6, [r0], #1
+    and r5, r6, #0xf
     mov r12, r5, lsl #8
     ldrb r6, [r0], #1
     orr r5, r6, r12
@@ -151,7 +106,7 @@ asm void MI_UncompressLZ8 (register const void * srcp, register void * destp)
     bgt @22
     b @21
 @26:
-    ldmfd sp !, {r4 - r7, lr}
+    ldmfd sp !, {r4 - r6, lr}
     bx lr
 }
 
