diff --git a/libraries/base/src/dwc_connectinet.c b/libraries/base/src/dwc_connectinet.c
index 65160f7..af17196 100644
--- a/libraries/base/src/dwc_connectinet.c
+++ b/libraries/base/src/dwc_connectinet.c
@@ -352,12 +352,12 @@ BOOL DWC_UpdateConnection (void)
     if (DWCi_CheckDisconnected()) {
         u8 aptype = DWC_AC_GetApType();
 
-        if (aptype == DWC_AC_AP_TYPE_FALSE || aptype > DWC_AC_AP_TYPE_UNKNOWN) {
+        if (aptype > DWC_AC_AP_TYPE_UNKNOWN) {
             aptype = DWC_AC_AP_TYPE_UNKNOWN;
         }
 
         DWCi_SetError(DWC_ERROR_DISCONNECTED, -54000 - aptype);
-        DWC_Printf(DWC_REPORTFLAG_DEBUG, "Detected DCF shutdown.\n");
+        DWC_CleanupInet();
 
         return TRUE;
     }
